File: Python/factorial.py
# Function to calculate the factorial recursively
def factorial(val):
    if val == 0:
        return 1
    else:
        return val * factorial(val-1)

#Prompting user for input
num = int(input("Enter the non-negative integer whose factorial is required: "))

#Check for non-negative integer
if num < 0:
    print("Invalid input! Entered integer is less than 0.")
else:
    print(factorial(num))

File: C#/factorial.cs
using System;  
public class Exercise15  
{  
    public static void Main()
{
  int i,f=1,num;
  
	Console.Write("\n\n");
    Console.Write("Calculate the factorial of a given number:\n");
    Console.Write("--------------------------------------------");
    Console.Write("\n\n");  

  Console.Write("Input the number : ");
   num= Convert.ToInt32(Console.ReadLine()); 
  for(i=1;i<=num;i++)
      f=f*i;

  Console.Write("The Factorial of {0} is: {1}\n",num,f);
 }
}
